frontend/src/lib/components/PasswordStrength.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { Progress } from 'antd'
+import { green, orange, red } from '@ant-design/colors'
+
+const PASSWORD_TEST_REGEXES: RegExp[] = [
+    /.{8,}/, // at least 8 characters
+    /.{12,}/, // bonus if longer than 12 characters
+    /[a-z]/, // at least one lowercase letter
+    /[A-Z]/, // at least one uppercase letter
+    /\d/, // at least one number
+    /[^A-Za-z0-9]/, // at least one special character
+]
+
+function scorePassword(password?: string): number {
+    if (!password) return 0
+    const testsPassed: number = PASSWORD_TEST_REGEXES.reduce(
+        (accumulator: number, regex: RegExp) => accumulator + Number(regex.test(password)),
+        0
+    )
+    return (testsPassed / PASSWORD_TEST_REGEXES.length) * 100
```

Comment:
Hmm... I would *nit* here a bit since this `reduce` seems rather magical.

Converting a `boolean` to a `number` like that seems a bit like alchemy ⚗️ ... and I think it's usually best to avoid the more complicated `reduce` in favor of simpler `map` and `filter` operations. 

Something like `const testsPassed = PASSWORD_TEST_REGEXES.map(r => r.test(password)).filter(p => p).length` reads cleaner IMO.

Fine to keep it as though :).